src/find-cycles/find-cycles.cabal

```diff
@@ -0,0 +1,15 @@
+cabal-version: >= 1.8
+version: 0.2023.10.3
+name:       find-cycles
+synopsis:   Script to find cross-module declaration cycles
+author:     Lawrence Chonavel
+build-type: Simple
+
+executable find-cycles
+  main-is:          FindCycles.hs
+  build-depends:
+      algebraic-graphs    >= 0.6.1 && < 0.7
+    , base                >= 4.16.4 && < 4.17
+    , containers          >= 0.6.5 && < 0.7
+    , hiedb               >= 0.4.3 && < 0.5
+    , temporary           >= 1.3 && < 1.4
```

Comment:
Could you please also add a README to `src/find-cycles` that explains how the tool is run (including steps like the hie generation if needed).

I tried to build `find-cycles` but had to add a `cabal.project` with `packages: .`.  Then it worked with `cabal build -w ghc-9.2.8`.